x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -1137,6 +1144,53 @@ private LogicalPlan rule(Eval eval) {
         }
     }
 
+    /**
+     * The vast majority of aggs ignore null entries - this rule adds a filter to filter this entries out
+     * to begin with.
+     * STATS x = min(a), y = sum(b)
+     * becomes
+     * | WHERE a IS NOT NULL OR b IS NOT NULL
```

Comment:
If we aren't pushing down the not-null constraints we shouldn't make the EVAL bit. Or we can make it, but then it should get folded into the agg. The agg ignores the null all on it's own.

Reply:
Moved the filter to the local optimizer so it gets added only when there's a local agg - otherwise it gets skipped.
This shows up in the tests where if an agg is on the coordinator (due to a limit for example), the rule doesn't kick in.